tools/c7n_oci/tests/oci_common.py

```diff
@@ -8,12 +8,12 @@
 
 from c7n.config import Config
 from c7n.schema import generate
-from c7n.testing import C7N_FUNCTIONAL, CustodianTestCore
+from c7n.testing import C7N_FUNCTIONAL, PyTestUtils
 
 FILTERED_FIELDS = ["metadata"]
 
 
-class OciBaseTest(CustodianTestCore):
+class OciBaseTest(PyTestUtils):
```

Comment:
Hey @kapilt , after this change the pytest runner is skipping a majority of the tests because they now contain the __init__ method. I tried to remedy this by trying a variety of ways like using a metaclass or using fixtures but unfortunately it did not work. The one way it would have worked was to override the PyCollector class and customizing the collect method, but I am not sure if such an extreme approach is acceptable. Another approach I thought of was to use the GCP way where the tests functions are not tied to a class but then I will have to end up making a copy of CustodianTestCore.capture_logging method. I'm not sure what will be the preferred approach in dealing with this. Please let me know your thoughts

Reply:
which `__init__` method are you referencing. I'm still not clear on why we're seeing an actual issue with the current setup.

re fixing, i think i see the confusion, we’re using the same testing.py class as a fixture (conftest) and as a base class for tests (oci_common). usage of addcleanup in the later is problematic, and tests need to be fixed to use the fixture.

afaics most of the tests are passing in the test fixture explicitly, so to disambiguate its a question of why we are using the fixture base class as a test base class, and then removing that from the base class.

Reply:
Updated according to our discussion earlier today